src/task/blocking.rs

```diff
@@ -47,18 +58,30 @@ lazy_static! {
     };
 }
 
-// Create up to MAX_THREADS dynamic blocking task worker threads.
-// Dynamic threads will terminate themselves if they don't
-// receive any work after between one and ten seconds.
-fn maybe_create_another_blocking_thread() {
-    // We use a `Relaxed` atomic operation because
-    // it's just a heuristic, and would not lose correctness
-    // even if it's random.
-    let workers = DYNAMIC_THREAD_COUNT.load(Ordering::Relaxed);
-    if workers >= MAX_THREADS {
-        return;
+fn calculate_dispatch_frequency() {
+    // Calculate current message processing rate here
+    let previous_freq = FREQUENCY.fetch_sub(1, Ordering::Relaxed);
+    let avr_freq = AVR_FREQUENCY.load(Ordering::Relaxed);
+    let current_pool_size = CURRENT_POOL_SIZE.load(Ordering::Relaxed);
+    let frequency = (avr_freq as f64 + previous_freq as f64 / current_pool_size as f64) as u64;
```

Comment:
Should the avr and previous freq's be collectively divided by the current pool size? It's not totally clear to me how this avoids creating a feedback loop.

Reply:
Actually, that holds the current one. I made it more clear.

Reply:
That's how moving average works in general. It won't create a feedback loop since finished threads will decrease the needed count and that saturating thread join will join them over time. It won't spawn new threads, because current pool size is controlling thread creation.